### PR TITLE
Feat: add date range

### PR DESCRIPTION
**Pull request by Rrrricky**
This PR adds an optional date range picker on the event creation modal and event update page, to set a range of bookable dates in the calendar.
A minDate and maxDate entries were added to the prisma schema to have a CRUD on each event.

### CODE REVIEW

File: pages/[user]/[type].tsx [hunk 1]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+        }
+    });
 
-  if (!user) {
-    return {
-      notFound: true,
-    };
-  }
+    const subEventTypes = user.eventTypes.map(el => {
+      if (el.startDate !== null) {
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:272]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
    if (el.startDate !== null) {
```

File: pages/[user]/[type].tsx [hunk 2]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+        return {
+          startDate: el.startDate.toString(),
+          endDate: el.endDate.toString(),
+        }
+      }
+      else {
+        return { 
+          startDate: el.startDate,
+          endDate: el.endDate,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:281]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
        endDate: el.endDate,
```

File: pages/[user]/[type].tsx [hunk 3]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+            id: true,
+            username: true,
+            name: true,
+            email: true,
+            bio: true,
+            avatar: true,
+            eventTypes: true,
+            startTime: true,
+            timeZone: true,
+            endTime: true,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:266]: Codacy has a fix for the issue: [Replace `············` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      endTime: true,
```

File: pages/success.tsx [hunk 4]
```diff
@@ -157,9 +157,28 @@ export async function getServerSideProps(context) {
         }
     });
 
+    const subEventTypes = user.eventTypes.map(el => {
+        if (el.startDate !== null) {
+          return {
```
Comment (baileypumfleet) [inline pages/success.tsx:162]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
  return {
```

File: pages/[user]/[type].tsx [hunk 5]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+export async function getServerSideProps(context) {
+    const user = await prisma.user.findFirst({
+        where: {
+          username: context.query.user,
+        },
+        select: {
+            id: true,
+            username: true,
+            name: true,
+            email: true,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:260]: Codacy has a fix for the issue: [Replace `············` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      email: true,
```

File: pages/[user]/[type].tsx [hunk 6]
```diff
@@ -76,25 +81,33 @@ export default function Type(props): Type {
...
 
+  const checkAvailableDays = (day) => {
+    if (selectedMonth > minMonth && selectedMonth < maxMonth) {
+      return true
+    }
+    else if (selectedMonth > minMonth && selectedMonth === maxMonth) {
+      return day <= maxDay
+    }
+    else if (selectedMonth === minMonth && selectedMonth < maxMonth) {
+      return day >= minDay
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:92]: Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      return day >= minDay;
```

File: pages/[user]/[type].tsx [hunk 7]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+            weekStart: true,
+        }
+    });
 
-  if (!user) {
-    return {
-      notFound: true,
-    };
-  }
+    const subEventTypes = user.eventTypes.map(el => {
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:271]: Codacy has a fix for the issue: [Replace `····const·subEventTypes·=·user.eventTypes.map(el` with `··const·subEventTypes·=·user.eventTypes.map((el)`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
  const subEventTypes = user.eventTypes.map((el) => {
```

File: pages/[user]/[type].tsx [hunk 8]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+      if (el.startDate !== null) {
+        return {
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:273]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      return {
```

File: pages/[user]/[type].tsx [hunk 9]
```diff
@@ -37,6 +37,11 @@ export default function Type(props): Type {
   const [timeFormat, setTimeFormat] = useState("h:mma");
   const telemetry = useTelemetry();
 
+  const minDay = parseInt(dayjs(props.eventType.startDate).format('D')) || null
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:40]: Codacy has a fix for the issue: [Replace `'D'))·||·null` with `"D"))·||·null;`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
  const minDay = parseInt(dayjs(props.eventType.startDate).format("D")) || null;
```

File: pages/[user]/[type].tsx [hunk 10]
```diff
@@ -176,8 +189,15 @@ export default function Type(props): Type {
                     disabled={selectedMonth < parseInt(dayjs().format("MM"))}>
                     <ChevronLeftIcon className="w-5 h-5" />
                   </button>
-                  <button onClick={incrementMonth}>
-                    <ChevronRightIcon className="w-5 h-5" />
+                  <button
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:192]: Codacy has a fix for the issue: [Delete `·`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
                  <button
```

File: pages/[user]/[type].tsx [hunk 11]
```diff
@@ -37,6 +37,11 @@ export default function Type(props): Type {
   const [timeFormat, setTimeFormat] = useState("h:mma");
   const telemetry = useTelemetry();
 
+  const minDay = parseInt(dayjs(props.eventType.startDate).format('D')) || null
+  const maxDay = parseInt(dayjs(props.eventType.endDate).format('D')) || null
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:41]: Codacy has a fix for the issue: [Replace `'D'))·||·null` with `"D"))·||·null;`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
  const maxDay = parseInt(dayjs(props.eventType.endDate).format("D")) || null;
```

File: pages/[user]/[type].tsx [hunk 12]
```diff
@@ -37,6 +37,11 @@ export default function Type(props): Type {
...
+  const minMonth = parseInt(dayjs(props.eventType.startDate).format('MM')) - 1 || null
+  const maxMonth = parseInt(dayjs(props.eventType.endDate).format('MM')) - 1 || null
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:43]: Codacy has a fix for the issue: [Replace `'MM'))·-·1·||·null` with `"MM"))·-·1·||·null;`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
  const maxMonth = parseInt(dayjs(props.eventType.endDate).format("MM")) - 1 || null;
```

File: pages/[user]/[type].tsx [hunk 13]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+          username: context.query.user,
+        },
+        select: {
+            id: true,
+            username: true,
+            name: true,
+            email: true,
+            bio: true,
+            avatar: true,
+            eventTypes: true,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:263]: Codacy has a fix for the issue: [Delete `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      eventTypes: true,
```

File: pages/[user]/[type].tsx [hunk 14]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
-      eventTypes: true,
-      startTime: true,
-      timeZone: true,
-      endTime: true,
-      weekStart: true,
-    },
-  });
+export async function getServerSideProps(context) {
+    const user = await prisma.user.findFirst({
+        where: {
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:253]: Codacy has a fix for the issue: [Delete `····`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
    where: {
```

File: pages/[user]/[type].tsx [hunk 15]
```diff
@@ -76,25 +81,33 @@ export default function Type(props): Type {
...
-        className={
-          "text-center w-10 h-10 rounded-full mx-auto " +
-          (dayjs().isSameOrBefore(dayjs().date(day).month(selectedMonth))
-            ? "bg-blue-50 text-blue-600 font-medium"
-            : "text-gray-400 font-light") +
-          (dayjs(selectedDate).month(selectedMonth).format("D") == day
-            ? " bg-blue-600 text-white-important"
-            : "")
-        }>
+        disabled={selectedMonth < parseInt(dayjs().format('MM')) && dayjs().month(selectedMonth).format('D') > day || minDay !== null ? !checkAvailableDays(day) : '' }
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:109]: Codacy has a fix for the issue: [Replace `selectedMonth·<·parseInt(dayjs().format('MM'))·&&·dayjs().month(selectedMonth).format('D')·>·day·||·minDay·!==·null·?·!checkAvailableDays(day)·:·''·}·` with `⏎··········(selectedMonth·<·parseInt(dayjs().format("MM"))·&&⏎············dayjs().month(selectedMonth).format("D")·>·day)·||⏎··········minDay·!==·null⏎············?·!checkAvailableDays(day)⏎············:·""⏎········}`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
        disabled={
```

File: pages/[user]/[type].tsx [hunk 16]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
-      startTime: true,
-      timeZone: true,
-      endTime: true,
-      weekStart: true,
-    },
-  });
+export async function getServerSideProps(context) {
+    const user = await prisma.user.findFirst({
+        where: {
+          username: context.query.user,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:254]: Codacy has a fix for the issue: [Replace `··········` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      username: context.query.user,
```

File: pages/[user]/[type].tsx [hunk 17]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+            email: true,
+            bio: true,
+            avatar: true,
+            eventTypes: true,
+            startTime: true,
+            timeZone: true,
+            endTime: true,
+            weekStart: true,
+        }
+    });
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:269]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
  });
```

File: pages/[user]/[type].tsx [hunk 18]
```diff
@@ -76,25 +81,33 @@ export default function Type(props): Type {
     setSelectedDate(dayjs().month(selectedMonth).date(day));
   };
 
+  const checkAvailableDays = (day) => {
+    if (selectedMonth > minMonth && selectedMonth < maxMonth) {
+      return true
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:86]: Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      return true;
```

File: pages/[user]/[type].tsx [hunk 19]
```diff
@@ -76,25 +81,33 @@ export default function Type(props): Type {
...
-          "text-center w-10 h-10 rounded-full mx-auto " +
-          (dayjs().isSameOrBefore(dayjs().date(day).month(selectedMonth))
-            ? "bg-blue-50 text-blue-600 font-medium"
-            : "text-gray-400 font-light") +
-          (dayjs(selectedDate).month(selectedMonth).format("D") == day
-            ? " bg-blue-600 text-white-important"
-            : "")
-        }>
+        disabled={selectedMonth < parseInt(dayjs().format('MM')) && dayjs().month(selectedMonth).format('D') > day || minDay !== null ? !checkAvailableDays(day) : '' } 
+        className={"text-center w-10 h-10 rounded-full mx-auto " + (minDay === null ? dayjs().isSameOrBefore(dayjs().date(day).month(selectedMonth)) ? 'bg-blue-50 text-blue-600 font-medium' : 'text-gray-400 font-light' : checkAvailableDays(day) ? 'bg-blue-50 text-blue-600 font-medium' : 'text-gray-400 font-light') + (minDay !== null ? (day >= minDay && day <= maxDay && dayjs(selectedDate).month(selectedMonth).format("D") == day ? ' bg-blue-600 text-white-important' : '') : (dayjs(selectedDate).month(selectedMonth).format("D") == day ? ' bg-blue-600 text-white-important' : undefined))}>
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:110]: Codacy has a fix for the issue: [Replace `"text-center·w-10·h-10·rounded-full·mx-auto·"·+·(minDay·===·null·?·dayjs().isSameOrBefore(dayjs().date(day).month(selectedMonth))·?·'bg-blue-50·text-blue-600·font-medium'·:·'text-gray-400·font-light'·:·checkAvailableDays(day)·?·'bg-blue-50·text-blue-600·font-medium'·:·'text-gray-400·font-light')·+·(minDay·!==·null·?·(day·>=·minDay·&&·day·<=·maxDay·&&·dayjs(selectedDate).month(selectedMonth).format("D")·==·day·?·'·bg-blue-600·text-white-important'·:·'')·:·(dayjs(selectedDate).month(selectedMonth).fo](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
        className={
```

File: pages/[user]/[type].tsx [hunk 20]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
-      endTime: true,
-      weekStart: true,
-    },
-  });
+export async function getServerSideProps(context) {
+    const user = await prisma.user.findFirst({
+        where: {
+          username: context.query.user,
+        },
+        select: {
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:256]: Codacy has a fix for the issue: [Delete `····`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
    select: {
```

File: pages/[user]/[type].tsx [hunk 21]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+            id: true,
+            username: true,
+            name: true,
+            email: true,
+            bio: true,
+            avatar: true,
+            eventTypes: true,
+            startTime: true,
+            timeZone: true,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:265]: Codacy has a fix for the issue: [Replace `············` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      timeZone: true,
```

File: pages/[user]/[type].tsx [hunk 22]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+    const subEventTypes = user.eventTypes.map(el => {
+      if (el.startDate !== null) {
+        return {
+          startDate: el.startDate.toString(),
+          endDate: el.endDate.toString(),
+        }
+      }
+      else {
+        return { 
+          startDate: el.startDate,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:280]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
        startDate: el.startDate,
```

File: pages/[user]/[type].tsx [hunk 23]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
-    },
-  });
+export async function getServerSideProps(context) {
+    const user = await prisma.user.findFirst({
+        where: {
+          username: context.query.user,
+        },
+        select: {
+            id: true,
+            username: true,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:258]: Codacy has a fix for the issue: [Replace `············` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      username: true,
```

File: pages/[user]/[type].tsx (hunk 2 above)
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:281]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
        endDate: el.endDate,
```

File: pages/[user]/[type].tsx [hunk 24]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+            name: true,
+            email: true,
+            bio: true,
+            avatar: true,
+            eventTypes: true,
+            startTime: true,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:264]: Codacy has a fix for the issue: [Replace `············` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      startTime: true,
```

File: pages/[user]/[type].tsx [hunk 25]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
-  });
+export async function getServerSideProps(context) {
+    const user = await prisma.user.findFirst({
+        where: {
+          username: context.query.user,
+        },
+        select: {
+            id: true,
+            username: true,
+            name: true,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:259]: Codacy has a fix for the issue: [Replace `············` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      name: true,
```

File: pages/[user]/[type].tsx (hunk 8 above)
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:273]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      return {
```

File: pages/availability/event/[type].tsx [hunk 26]
```diff
@@ -35,6 +36,17 @@ export default function EventType(props) {
     const [ locations, setLocations ] = useState(props.eventType.locations || []);
     const [ customInputs, setCustomInputs ] = useState<EventTypeCustomInput[]>(props.eventType.customInputs.sort((a, b) => a.id - b.id) || []);
     const locationOptions = props.locationOptions
+    const [ dates, setDates] = useState({ startDate: props.eventType.startDate !== null ? dayjs(new Date(props.eventType.startDate)) : dayjs(new Date()), endDate: props.eventType.endDate !== null ? dayjs(new Date(props.eventType.endDate)) : dayjs(new Date()) });
+    const [ timeRange, setTimeRange ] = useState<'unlimited' | 'specific' | undefined>();
```
Comment (baileypumfleet) [inline pages/availability/event/[type].tsx:40]: Codacy has a fix for the issue: [Replace `····const·[·timeRange,·setTimeRange·]·=·useState<'unlimited'·|·'specific'` with `··const·[timeRange,·setTimeRange]·=·useState<"unlimited"·|·"specific"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
  const [timeRange, setTimeRange] = useState<"unlimited" | "specific" | undefined>();
```

File: pages/[user]/[type].tsx (hunk 10 above)
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:192]: Codacy has a fix for the issue: [Delete `·`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
                  <button
```

File: pages/availability/event/[type].tsx [hunk 27]
```diff
@@ -35,6 +36,17 @@ export default function EventType(props) {
...
+
+    useEffect(() => {
+        if (props.eventType.startDate !== null) {
+            setTimeRange('specific');
+        }
+        else {
+            setTimeRange('unlimited');
+        }
+    }, [])
```
Comment (baileypumfleet) [inline pages/availability/event/[type].tsx:49]: Codacy has a fix for the issue: [Replace `····},·[])` with `··},·[]);`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
  }, []);
```

Comment (baileypumfleet) [inline pages/availability/event/[type].tsx:49]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'props.eventType.startDate'. Either include it or remove the dependency array.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
    }, [props.eventType.startDate])
```

File: pages/[user]/[type].tsx [hunk 28]
```diff
@@ -176,8 +189,15 @@ export default function Type(props): Type {
...
-                    <ChevronRightIcon className="w-5 h-5" />
+                  <button 
+                    onClick={incrementMonth}
+                    className={
+                      minDay !== null &&
+                      selectedMonth >= maxMonth &&
+                        "text-gray-400"
+                    }
+                    disabled={minDay !== null && selectedMonth >= maxMonth}>
+                      <ChevronRightIcon className="w-5 h-5" />
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:200]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
                    <ChevronRightIcon className="w-5 h-5" />
```

File: pages/[user]/[type].tsx (hunk 12 above)
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:43]: Codacy has a fix for the issue: [Replace `'MM'))·-·1·||·null` with `"MM"))·-·1·||·null;`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
  const maxMonth = parseInt(dayjs(props.eventType.endDate).format("MM")) - 1 || null;
```

File: pages/[user]/[type].tsx (hunk 13 above)
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:263]: Codacy has a fix for the issue: [Delete `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      eventTypes: true,
```

File: pages/[user]/[type].tsx (hunk 14 above)
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:253]: Codacy has a fix for the issue: [Delete `····`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
    where: {
```

File: pages/[user]/[type].tsx [hunk 29]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+            username: true,
+            name: true,
+            email: true,
+            bio: true,
+            avatar: true,
+            eventTypes: true,
+            startTime: true,
+            timeZone: true,
+            endTime: true,
+            weekStart: true,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:267]: Codacy has a fix for the issue: [Replace `············` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      weekStart: true,
```

File: pages/[user]/[type].tsx (hunk 15 above)
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:109]: Codacy has a fix for the issue: [Replace `selectedMonth·<·parseInt(dayjs().format('MM'))·&&·dayjs().month(selectedMonth).format('D')·>·day·||·minDay·!==·null·?·!checkAvailableDays(day)·:·''·}·` with `⏎··········(selectedMonth·<·parseInt(dayjs().format("MM"))·&&⏎············dayjs().month(selectedMonth).format("D")·>·day)·||⏎··········minDay·!==·null⏎············?·!checkAvailableDays(day)⏎············:·""⏎········}`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
        disabled={
```

File: pages/[user]/[type].tsx (hunk 16 above)
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:254]: Codacy has a fix for the issue: [Replace `··········` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      username: context.query.user,
```

File: pages/availability/index.tsx [hunk 30]
```diff
@@ -249,6 +253,59 @@ export default function Availability(props) {
...
+                                                    <input
+                                                        type="radio"
+                                                        className="form-radio"
+                                                        onChange={() => setTimeRange('specific')}
+                                                        name="dateRange"
+                                                        value="specific"
+                                                        checked={timeRange === 'specific'}
+                                                    />
+                                                    <span className="ml-2 text-xs">Range</span>
+                                                </label>
```
Comment (baileypumfleet) [inline pages/availability/index.tsx:273]: Codacy has a fix for the issue: [Delete `························`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
                        </label>
```

File: pages/availability/event/[type].tsx [hunk 31]
```diff
@@ -13,6 +13,7 @@ import {
   XIcon,
   PhoneIcon,
 } from '@heroicons/react/outline';
+import dayjs from 'dayjs';
```
Comment (baileypumfleet) [inline pages/availability/event/[type].tsx:16]: Codacy has a fix for the issue: [Replace `'dayjs'` with `"dayjs"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
import dayjs from "dayjs";
```

File: pages/[user]/[type].tsx (hunk 18 above)
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:86]: Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      return true;
```

File: pages/[user]/[type].tsx (hunk 20 above)
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:256]: Codacy has a fix for the issue: [Delete `····`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
    select: {
```

File: pages/[user]/[type].tsx [hunk 32]
```diff
@@ -76,25 +81,33 @@ export default function Type(props): Type {
...
+      return day <= maxDay
+    }
+    else if (selectedMonth === minMonth && selectedMonth < maxMonth) {
+      return day >= minDay
+    }
+    else if (selectedMonth === minMonth  && selectedMonth === maxMonth) {
+      return day >= minDay && day <= maxDay
+    }
+    else {
+      return false
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:98]: Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      return false;
```

File: pages/[user]/[type].tsx [hunk 33]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
+    const user = await prisma.user.findFirst({
+        where: {
+          username: context.query.user,
+        },
+        select: {
+            id: true,
+            username: true,
+            name: true,
+            email: true,
+            bio: true,
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:261]: Codacy has a fix for the issue: [Delete `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      bio: true,
```

File: pages/[user]/[type].tsx [hunk 34]
```diff
@@ -228,57 +248,84 @@ export default function Type(props): Type {
...
 
-  if (!user) {
-    return {
-      notFound: true,
-    };
-  }
+    const subEventTypes = user.eventTypes.map(el => {
+      if (el.startDate !== null) {
+        return {
+          startDate: el.startDate.toString(),
```
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:274]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
        startDate: el.startDate.toString(),
```

File: pages/[user]/[type].tsx (hunk 23 above)
Comment (baileypumfleet) [inline pages/[user]/[type].tsx:258]: Codacy has a fix for the issue: [Replace `············` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689307)
```suggestion
      username: true,
```

**Comment by emrysal** [comment]
@Rrrricky This has been superseded now by advanced availability - from this branch originated the idea for superjson which unfortunately breaks the login page (for some reason) when installing it. I hope you find the availability scheduler useful (part of all event type pages).

I've produced a walkthrough / feature demo too: https://www.loom.com/share/e1ef9b4ca0a740809f8a961a79c1aab3

**Pull request closed without merging**